src/org/zaproxy/zap/extension/pscanrulesBeta/CookieLooselyScopedScanner.java

```diff
@@ -83,12 +83,12 @@ private boolean isLooselyScopedCookie(HttpCookie cookie, String host) {
 		// preconditions
 		assert cookie != null;
 		assert host != null;
-		
+
 		String cookieDomain = cookie.getDomain();
-		
+
 		// if Domain attribute hasn't been specified, the cookie
 		// is scoped with the response host
-		if (cookieDomain == null) {
+		if (cookieDomain == null || cookieDomain.equalsIgnoreCase("null") || cookieDomain.isEmpty()) {

```

Comment:
I must have removed the test for that edge case somehow, but one way to reproduce it is `Set-Cookie: a=b;domain=null` or having a malformed domain `Set-Cookie: a=b;domain`.
I will write down that edge case, if I remember it..